navigator_engine/pluggable_logic/conditional_functions.py

```diff
@@ -81,9 +81,34 @@ def check_spectrum_file(indicators: list[str], engine: DecisionEngine) -> bool:
                 f'Indicator "{indicator}" not found in Spectrum checklist'
             )
     indicator_checks = checklist[checklist['ID'].isin(indicators)]
-    indicator_checks['Status'] = indicator_checks['Status'].replace(
-        [0, 'FALSE', 'F', 'false', 'f'],
-        value=False,
-    )
+    indicator_checks['Status'].replace([0, 'FALSE', 'F', 'false', 'f'],
+                                       value=False,
+                                       inplace=True)
+
     indicator_checks['Status'][indicator_checks['Status'].ne(False)] = True
+
     return indicator_checks['Status'].eq(True).all()
+
+
+@register_conditional
+def check_naomi_file(indicators: list[str], engine: DecisionEngine) -> bool:
+    checklist = engine.data['naomi-validation-file']['data']
+
+    if checklist is None:
+        return False
+
+    for indicator in indicators:
+        if indicator not in checklist['NaomiCheckPermPrimKey'].values:
```

Comment:
Can we make this case insensitive?
```suggestion
    indicators = [indicator.lower() for indicator in indicators]
    for indicator in indicators:
        if indicator not in checklist['NaomiCheckPermPrimKey'].str.lower().values:
```